src/drt/src/pa/FlexPA_row_pattern.cpp

```diff
@@ -63,6 +63,60 @@ std::vector<std::vector<frInst*>> FlexPA::computeInstRows()
   return inst_rows;
 }
 
+bool FlexPA::instancesAreAbuting(frInst* inst_1, frInst* inst_2) const
+{
+  if (inst_1->getOrigin().getY() != inst_2->getOrigin().getY()) {
+    return false;
+  }
+  frInst *left_inst, *right_inst;
+  if (inst_1->getOrigin().getX() < inst_2->getOrigin().getX()) {
+    left_inst = inst_1;
+    right_inst = inst_2;
+  } else {
+    left_inst = inst_2;
+    right_inst = inst_1;
+  }
+
+  if (left_inst->getBoundaryBBox().xMax() != right_inst->getOrigin().getX()) {
+    return false;
+  }
```

Comment:
Could we rely on the bbox instead of the origin in this code? There is a reverted PR that uses the instance's origin correctly which would make this incorrect (origin may be the center of the bbox not lower left). I am planning to rework this PR soon.